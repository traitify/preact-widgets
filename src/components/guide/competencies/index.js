import PropTypes from "prop-types";
import {Component} from "react";
import QuestionSequences from "../question-sequences/index";

class Competencies extends Component {
  static defaultProps = {competencies: null}
  static propTypes = {
    competencies: PropTypes.arrayOf(PropTypes.object)
  }
  constructor(props) {
    super(props);

    this.state = {
      competencies: props.competencies,
      displayedCompetency: props.competencies[0]
    };
  }
  displayedCompetency(competency) {
    this.state.competencies.forEach((comp) => {
      if(competency.target.innerText === comp.id) {
        this.setState({displayedCompetency: comp});
      }
    });
  }
  render() {
    const {name, id, introduction, questionSequences} = this.state.displayedCompetency;
    return (
      <div className="competency-container">
        {this.state.competencies.map((competency) => (
          <div id="badge-container" key={competency.id}>
            <div role="link" tabIndex={0} onKeyPress={this.displayedCompetency.bind(this)} onClick={this.displayedCompetency.bind(this)}>{competency.id}</div>
          </div>
        ))}
        <div className="competency" key={id}>
          <div className="name">{name}</div>
          <div className="introduction">{introduction}</div>
          <div className="question-sequences">
            <QuestionSequences questionSequences={questionSequences} />
          </div>
          <br />
        </div>
      </div>
    );
  }
}

export default Competencies;
