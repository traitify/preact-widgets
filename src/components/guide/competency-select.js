import PropTypes from "prop-types";
import {Component} from "react";
import style from "./style.scss";

export default class CompetencySelect extends Component {
  static propTypes = {
    competencies: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.string.isRequired,
        name: PropTypes.string.isRequired
      })
    ).isRequired,
    displayedCompetency: PropTypes.shape({
      id: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired
    }).isRequired,
    displayCompetency: PropTypes.func.isRequired,
    tabBadge: PropTypes.func.isRequired
  }
  displayCompetency = (e) => {
    this.props.displayCompetency(e.target.value);
  }
  render() {
    const {competencies, displayedCompetency, tabBadge} = this.props;

    return (
      <div className={style.competencySelect}>
        <select
          value={displayedCompetency.id}
          className={style.mobileSelect}
          onChange={this.displayCompetency}
        >
          {competencies.map((competency) => (
            <option key={competency.id} value={competency.id}>{competency.name}</option>
          ))}
        </select>
        <p className={style.mobileBadge}>
          <img src={tabBadge(displayedCompetency.id)} alt={`${displayedCompetency.name} badge`} />
        </p>
      </div>
    );
  }
}
