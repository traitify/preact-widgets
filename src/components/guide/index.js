import PropTypes from "prop-types";
import {Component} from "react";
import DangerousHTML from "lib/helpers/dangerous-html";
import TraitifyPropTypes from "lib/helpers/prop-types";
import withTraitify from "lib/with-traitify";
import CompetencySelect from "./competency-select";
import CompetencyTab from "./competency-tab";
import style from "./style.scss";

class Guide extends Component {
  static defaultProps = {assessment: null, guide: null}
  static propTypes = {
    assessment: PropTypes.shape({
      assessment_type: PropTypes.string,
      personality_types: PropTypes.array
    }),
    followGuide: PropTypes.func.isRequired,
    guide: PropTypes.shape({
      assessment_id: PropTypes.string.isRequired,
      competencies: PropTypes.array.isRequired,
      locale_key: PropTypes.string.isRequired
    }),
    isReady: PropTypes.func.isRequired,
    translate: PropTypes.func.isRequired,
    ui: TraitifyPropTypes.ui.isRequired
  }
  constructor(props) {
    super(props);
    this.state = {
      badges: [],
      competencies: [],
      displayedCompetency: null,
      showExpandedIntro: false
    };
  }
  componentDidMount() {
    this.props.ui.trigger("Guide.initialized", this);
    this.props.followGuide();
  }
  componentDidUpdate(prevProps) {
    this.props.ui.trigger("Guide.updated", this);

    const guide = this.props.guide || {};
    const prevGuide = prevProps.guide || {};
    const changes = {
      assessment: guide.assessment_id !== prevGuide.assessment_id,
      locale: guide.locale_key !== prevGuide.locale_key
    };

    if(changes.assessment || changes.locale) { this.setGuide(); }
  }
  setGuide() {
    if(!this.props.guide) {
      return this.setState({badges: [], competencies: [], displayedCompetency: null});
    }

    const {competencies} = this.props.guide;
    const badges = this.badges(competencies);

    this.setState({badges, competencies, displayedCompetency: {...competencies[0]}});
  }
  displayCompetency = (competencyID) => {
    this.setState(({competencies}) => {
      const displayedCompetency = competencies.find((competency) => competency.id === competencyID);

      return displayedCompetency && {displayedCompetency};
    });
  }
  tabBadge = (id) => (
    this.state.badges.find((badge) => badge.competencyID === id).image
  )
  toggleExpandedIntro = () => {
    this.setState((prevState) => ({showExpandedIntro: !prevState.showExpandedIntro}));
  }
  badges(competencies) {
    const {assessment} = this.props;

    return competencies.map((competency) => {
      const personalityID = competency.questionSequences[0].personality_type_id;
      const types = assessment.personality_types;
      const personality = types.find((type) => type.personality_type.id === personalityID);

      return {competencyID: competency.id, image: personality.personality_type.badge.image_medium};
    });
  }
  introduction() {
    const {introduction} = this.state.displayedCompetency;
    const intro = introduction.split("\n", 1)[0];
    const readMore = introduction.replace(intro, "").trim();

    return {intro, readMore};
  }
  stringToListItems(entity) {
    let entities = entity.split("\n");
    /* eslint-disable-next-line react/no-array-index-key */
    entities = entities.map((e, i) => <li key={i}>{e}</li>);

    return (<ul>{entities}</ul>);
  }
  render() {
    if(!this.props.isReady("guide")) { return null; }
    if(!this.props.isReady("results")) { return null; }
    if(this.state.competencies.length === 0) { return null; }

    const {translate} = this.props;
    const {competencies, displayedCompetency, showExpandedIntro} = this.state;
    const {intro, readMore} = this.introduction();

    return (
      <div className={style.tabsContainer}>
        <div className={style.tabContainer}>
          <CompetencySelect
            competencies={competencies}
            displayedCompetency={displayedCompetency}
            displayCompetency={this.displayCompetency}
            tabBadge={this.tabBadge}
          />
          <ul className={style.tabs}>
            {competencies.map((competency) => (
              <CompetencyTab
                key={competency.id}
                competency={competency}
                displayedCompetency={displayedCompetency}
                displayCompetency={this.displayCompetency}
                tabBadge={this.tabBadge}
              />
            ))}
          </ul>
        </div>
        <div className={style.tabsContent}>
          <div className={style.tabContentActive}>
            <h2>{displayedCompetency.name}</h2>
            {intro}
            <p>
              <button
                type="button"
                onClick={this.toggleExpandedIntro}
              >
                {translate("read_more")}
              </button>
            </p>
            {showExpandedIntro && (<p>{readMore}</p>)}
            <hr />
            {displayedCompetency.questionSequences.map((sequence) => (
              <div key={sequence.id}>
                <h2>{sequence.name}</h2>
                <p>{translate("guide_intro")}</p>
                <p><DangerousHTML html={translate("guide_get_started_html")} tag="em" /></p>
                {sequence.questions.map((question) => (
                  <div key={question.id}>
                    <h3 id={question.order === 1 ? "traitify-question-1" : null}>{`Question ${question.order}`}</h3>
                    <p>{question.text}</p>
                    <h4>{translate("question_purpose")}</h4>
                    <div>{this.stringToListItems(question.purpose)}</div>
                    {question.adaptability && (
                      <div>
                        <h4>{translate("question_adaptability")}</h4>
                        <div>{this.stringToListItems(question.adaptability)}</div>
                      </div>
                    )}
                  </div>
                ))}
              </div>
            ))}
          </div>
        </div>
      </div>
    );
  }
}

export {Guide as Component};
export default withTraitify(Guide);
