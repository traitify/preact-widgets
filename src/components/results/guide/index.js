import {faChevronDown, faChevronUp} from "@fortawesome/free-solid-svg-icons";
import PropTypes from "prop-types";
import {useEffect, useState} from "react";
import {useDidMount, useDidUpdate} from "lib/helpers/hooks";
import Icon from "lib/helpers/icon";
import {dig} from "lib/helpers/object";
import TraitifyPropTypes from "lib/helpers/prop-types";
import withTraitify from "lib/with-traitify";
import style from "./style.scss";

const toList = (entity) => (
  /* eslint-disable-next-line react/no-array-index-key */
  <ul>{entity.split("\n").map((e, i) => <li key={i}>{e}</li>)}</ul>
);

function Question({question, translate}) {
  const [showContent, setShowContent] = useState(question.order === 1);

  return (
    <div className={style.question}>
      <h3 id={question.order === 1 ? "traitify-question-1" : null}>{`Question ${question.order}`}</h3>
      <button onClick={() => setShowContent(!showContent)} type="button">
        <span className={style.questionText}>{question.text}</span>
        <Icon className={style.questionIcon} icon={showContent ? faChevronUp : faChevronDown} />
      </button>
      {showContent && (
        <>
          <h4>{translate("question_purpose")}</h4>
          <div>{toList(question.purpose)}</div>
          {question.adaptability && (
            <div>
              <h4>{translate("question_adaptability")}</h4>
              <div>{toList(question.adaptability)}</div>
            </div>
          )}
        </>
      )}
    </div>
  );
}

Question.propTypes = {
  question: PropTypes.shape({
    adaptability: PropTypes.string,
    order: PropTypes.number.isRequired,
    purpose: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired
  }).isRequired,
  translate: PropTypes.func.isRequired
};

function Guide(props) {
  const {assessment, followGuide, guide, isReady, translate, ui} = props;
  const competencies = dig(guide, ["competencies"]) || [];
  const types = dig(assessment, ["personality_types"]) || [];
  const data = competencies.map((competency) => {
    if(types.length === 0) { return competency; }

    const typeID = competency.questionSequences[0].personality_type_id;
    const {personality_type: {badge}, score} = types
      .find(({personality_type: {id}}) => id === typeID);
    let rank;

    switch(true) {
      case score <= 3: rank = "low"; break;
      case score <= 6: rank = "medium"; break;
      default: rank = "high";
    }

    return {...competency, badge: badge.image_medium, rank, score};
  }, {}).sort(({score: scoreA}, {score: scoreB}) => scoreA - scoreB);
  const [activeCompetency, setActiveCompetency] = useState(data[0]);
  const [showExpandedIntro, setShowExpandedIntro] = useState(false);
  const state = {activeCompetency, showExpandedIntro};

  useDidMount(() => { ui.trigger("Guide.initialized", {props, state}); });
  useDidMount(() => { followGuide(); });
  useDidUpdate(() => { ui.trigger("Guide.updated", {props, state}); });
  useEffect(() => {
    if(data.length === 0) { return; }

    setActiveCompetency(data[0]);
  }, [dig(assessment, ["id"]), dig(guide, ["assessment_id"]), dig(guide, ["locale_key"])]);

  if(!isReady("guide")) { return null; }
  if(!isReady("results")) { return null; }
  if(!activeCompetency) { return null; }
  if(competencies.length === 0) { return null; }

  const showCompetency = (newID) => setActiveCompetency(data.find(({id}) => newID === id));
  const [intro, ...expandedIntro] = activeCompetency.introduction.split("\n");
  const onChange = ({target: {value}}) => showCompetency(value);

  return (
    <div className={style.tabsContainer}>
      <div className={style.tabContainer}>
        <div className={style.competencySelect}>
          <select className={style.mobileSelect} onChange={onChange} value={activeCompetency.id}>
            {data.map(({id, name}) => <option key={id} value={id}>{name}</option>)}
          </select>
          <p className={style.mobileBadge}>
            <img src={activeCompetency.badge} alt={`${activeCompetency.name} badge`} />
          </p>
        </div>
        <ul className={style.tabs}>
          {data.map(({badge, id, name, rank}) => {
            const classes = [
              style[rank],
              style.tab,
              id === activeCompetency.id && style.tabActive
            ].filter(Boolean);

            return (
              <li key={id} className={classes.join(" ")}>
                <button onClick={() => showCompetency(id)} type="button">
                  <img src={badge} alt={`${name} badge`} />
                  <br />
                  {name}
                </button>
              </li>
            );
          })}
        </ul>
      </div>
      <div className={style.tabsContent}>
        <div className={style.tabContentActive}>
          <h2>{activeCompetency.name}</h2>
          {intro}
          <p>
            <button onClick={() => setShowExpandedIntro(!showExpandedIntro)} type="button">
              {translate("read_more")}
            </button>
          </p>
          {showExpandedIntro && <p>{expandedIntro.join("\n").trim()}</p>}
          <hr />
          {activeCompetency.questionSequences.map((sequence) => (
            <div key={sequence.id}>
              <h2>{sequence.name}</h2>
              <p>{translate("guide_intro")}</p>
              {sequence.questions.map((question) => (
                <Question key={question.id} question={question} translate={translate} />
              ))}
            </div>
          ))}
        </div>
      </div>
    </div>
  );
}

Guide.defaultProps = {assessment: null, guide: null};
Guide.propTypes = {
  assessment: PropTypes.shape({
    assessment_type: PropTypes.string,
    personality_types: PropTypes.array
  }),
  followGuide: PropTypes.func.isRequired,
  guide: PropTypes.shape({
    assessment_id: PropTypes.string.isRequired,
    competencies: PropTypes.array.isRequired,
    locale_key: PropTypes.string.isRequired
  }),
  isReady: PropTypes.func.isRequired,
  translate: PropTypes.func.isRequired,
  ui: TraitifyPropTypes.ui.isRequired
};

export {Guide as Component};
export default withTraitify(Guide);
