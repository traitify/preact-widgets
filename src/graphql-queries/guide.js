import gql from "graphql-tag";

export default gql`
  query InterviewGuide($assessment_id: String!) {
    guide(assessmentId: $assessment_id, localeKey: "en-us") {
      id
      name
      deckId
      competencies {
        id
        name
        introduction
        order
        questionSequences {
          id
          name
          questions {
            id
            text
            adaptability
            order
          }
        }
      }
    }
  }
`;
