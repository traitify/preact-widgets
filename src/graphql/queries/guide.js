import GraphQL from "graphql/gql-client";

export default function guideQuery(guideDetails = {}) {
  let {params, fields} = guideDetails;

  params = Object.assign({localeKey: "en-US"}, params);

  const defaultFields = [
    "deckId", "id", "name",
    {competencies: ["id", "name", "introduction", "order", {questionSequences: ["id", "name", {questions: ["id", "text", "adaptability", "order", "purpose"]}]}]}
  ];
  fields = (fields === undefined ? defaultFields : fields);

  const graphql = new GraphQL();
  return `{ guide(${graphql.toArgs(params)}) { ${graphql.toQuery(fields)} }}`;
}
