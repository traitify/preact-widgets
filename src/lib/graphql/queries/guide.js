import toArgs from "../to-args";
import toQuery from "../to-query";

export default function guideQuery({fields: _fields, params}) {
  const fields = _fields || [
    {
      competencies: [
        "id", "name", "introduction", "order",
        {
          questionSequences: [
            "id", "name", "personality_type_id",
            {questions: ["id", "text", "adaptability", "order", "purpose"]}
          ]
        }
      ]
    }
  ];

  return `{ guide(${toArgs(params)}) { ${toQuery(fields)} }}`;
}
