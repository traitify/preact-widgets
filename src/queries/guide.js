import gql from "graphql-tag";

export default gql`
  query Guide($assessmentId: String!) {
    guide(assessmentId: $assessmentId, localeKey: "en-us") {
      id
      name
      deckId
      competencies {
        id
        name
        introduction
        order
        questionSequences {
          id
          name
          questions {
            id
            text
            adaptability
            order
          }
        }
      }
    }
  }
`;
