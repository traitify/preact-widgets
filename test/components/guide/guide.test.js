import {Component} from "components/guide/index";
import ComponentHandler from "support/component-handler";
import response from "support/json/guide.json";

describe("Guide", () => {
  let props;

  beforeEach(() => {
    props = {
      translate: jest.fn().mockName("translate").mockImplementation((value) => value),
      assessmentID: "bca3ab7c-8ed1-421f-a0ab-39d619efeee0",
      traitify: {
        graphqlQuery: jest.fn().mockName("graphqlQuery").mockImplementation(() => (Promise.resolve()))
      }
    };
  });

  it("formats introduction", () => {
    const component = (new ComponentHandler(<Component {...props} />));
    component.updateState(
      {
        competencies: response.data.guide.competencies,
        displayedCompetency: response.data.guide.competencies[0]
      }
    );
    const introduction = component.instance.introduction();

    expect(introduction.intro).toBe("At the top level, Solving Problems (O) concerns how someone thinks about work problems, projects\nand challenges, how receptive they are to new or different information or approaches, and the way this\ninfluences their decision making.");
    expect(introduction.readMore).toBe("The benefits of being open-minded are likely to include approaching problems in original ways, making\nquick connections, pushing boundaries, seeing the bigger picture, and not getting stuck in the detail.\nIn contrast someone who is pragmatic likes to solve problems in practical ways, and prefers tried-andtested solutions or established rules and regulations. They are likely to have a fine eye for detail, be results\noriented, and able to bring experience to bear on a problem. Those who are a mixture of both (openminded/pragmatic) are best described as approaching problems in a systematic way, being able to flex\nbetween different options, capable of seeing situations at different levels of detail, and as bringing healthy\ncaution to decision making.\nThese personality characteristics provide the basis for asking a particular sequence of questions of each\nof the personality 'types'. The first question is looking to confirm what they should be like; the second two\nexplore areas of possible concern.");
  });

  it("handles displayedCompetency", () => {
    const component = (new ComponentHandler(<Component {...props} />));
    component.updateState(
      {
        competencies: response.data.guide.competencies,
        displayedCompetency: response.data.guide.competencies[0]
      }
    );
    expect(component.state.displayedCompetency.name).toBe("Solving Problems");

    component.instance.displayCompetency("Delivering Results");
    expect(component.state.displayedCompetency.name).toBe("Delivering Results");

    component.instance.displayCompetency({target: {value: "Engaging with People"}});
    expect(component.state.displayedCompetency.name).toBe("Engaging with People");

    component.instance.displayCompetency("some invalid competency");
    expect(component.state.displayedCompetency.name).toBe("Engaging with People");
  });

  it("matches snapshot on success", () => {
    const component = (new ComponentHandler(<Component {...props} />));
    component.updateState(
      {
        competencies: response.data.guide.competencies,
        displayedCompetency: response.data.guide.competencies[0]
      }
    );

    expect(component.tree).toMatchSnapshot();
  });

  it("returns blank div on error", () => {
    const component = (new ComponentHandler(<Component {...props} />));
    component.updateState({errors: ["did not successfully fetch guide"]});
    expect(component.tree).toMatchSnapshot();
  });
});
