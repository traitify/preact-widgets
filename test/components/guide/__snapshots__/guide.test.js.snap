// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Guide matches snapshot on success 1`] = `
<div>
  <div>
    <ul>
      <li>
        <a
          href="#tab-0"
          name="Solving Problems"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={0}
        >
          <img
            alt="Solving Problems badge"
            src="https://cdn.traitify.com/assets/images/js/openness.png"
          />
          <br />
          Solving Problems
        </a>
      </li>
      <li
        className={null}
      >
        <a
          href="#tab-1"
          name="Delivering Results"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={0}
        >
          <img
            alt="Delivering Results badge"
            src="https://cdn.traitify.com/assets/images/js/conscientiousness.png"
          />
          <br />
          Delivering Results
        </a>
      </li>
      <li
        className={null}
      >
        <a
          href="#tab-2"
          name="Engaging with People"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={0}
        >
          <img
            alt="Engaging with People badge"
            src="https://cdn.traitify.com/assets/images/js/extraversion.png"
          />
          <br />
          Engaging with People
        </a>
      </li>
      <li
        className={null}
      >
        <a
          href="#tab-3"
          name="Influencing People"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={0}
        >
          <img
            alt="Influencing People badge"
            src="https://cdn.traitify.com/assets/images/js/agreeableness.png"
          />
          <br />
          Influencing People
        </a>
      </li>
      <li
        className={null}
      >
        <a
          href="#tab-4"
          name="Managing Pressure"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={0}
        >
          <img
            alt="Managing Pressure badge"
            src="https://cdn.traitify.com/assets/images/js/emotional_stability.png"
          />
          <br />
          Managing Pressure
        </a>
      </li>
    </ul>
  </div>
  <div>
    <div>
      <h2>
        Solving Problems
      </h2>
      At the top level, Solving Problems (O) concerns how someone thinks about work problems, projects
and challenges, how receptive they are to new or different information or approaches, and the way this
influences their decision making.
      <p>
        <a
          href="#read-more"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={0}
        >
          read_more
        </a>
      </p>
      <hr />
      <div
        className="competency-type"
      >
        <h2>
          Open-Minded/Pragmatic People
        </h2>
        <p>
          guide_intro
        </p>
        <p>
          <em
            dangerouslySetInnerHTML={
              Object {
                "__html": "guide_get_started_html",
              }
            }
          />
        </p>
        <div
          className="questions"
        >
          <h3
            id="traitify-question-1"
          >
            Question 1
          </h3>
          <p>
            Tell me how you would go about solving a work problem you have not encountered before.
          </p>
          <h4>
            question_purpose
          </h4>
          <div>
            <ul>
              <li>
                Do they talk about a number of different ways of approaching the problem?
              </li>
              <li>
                Are their ideas new, original or different in any way? Do they test new ideas against what seems feasible,
both now and in the longer term? Producing new ideas but checking their viability, and implementing
balanced and timely solutions, are all aspects of the open-minded/pragmatic character.
And to probe into possible issues with being open-minded/pragmatic:
              </li>
            </ul>
          </div>
        </div>
        <div
          className="questions"
        >
          <h3
            id={null}
          >
            Question 2
          </h3>
          <p>
            (Your results suggest you like to think about new ideas and possibilities). Give me an example
of a time when you had to be practical and realistic, and work to a fixed set of rules.
          </p>
          <h4>
            question_purpose
          </h4>
          <div>
            <ul>
              <li>
                How comfortable are they working in a step-by-step manner? Is there a
              </li>
              <li>
                tension between this and wanting to try something new? Does this 'conflict' sometimes show itself in taking
a very cautious approach?
              </li>
            </ul>
          </div>
          <div>
            <h4>
              question_adaptability
            </h4>
            <div>
              <ul>
                <li>
                  Do they recognize that sometimes the best approach is to use an
                </li>
                <li>
                  existing procedure, or to be far less cautious and try something radically different? Can they give an example
of using either of these extremes? What do they do to stop themselves getting stuck on the 'middle way' of
doing things?
                </li>
              </ul>
            </div>
          </div>
        </div>
        <div
          className="questions"
        >
          <h3
            id={null}
          >
            Question 3
          </h3>
          <p>
            (Your results suggest you are a grounded person but also someone who thinks about future
possibilities.) Can you describe a work situation where you realized that you hadn't learned from a similar
situation in the past. What happened and what did you do?
          </p>
          <h4>
            question_purpose
          </h4>
          <div>
            <ul>
              <li>
                Do they sometimes not learn from the past? Do they move on too quickly
              </li>
              <li>
                without consolidating what they have learned? Do they sometimes override their experience because they
think they need to demonstrate original thought?
              </li>
            </ul>
          </div>
          <div>
            <h4>
              question_adaptability
            </h4>
            <div>
              <ul>
                <li>
                  Do they know they sometimes confuse knowledge with foresight?
                </li>
                <li>
                  Can they provide an example of sticking with experience, and the tried-and-tested, even though they
wanted to try something fresh?
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Guide returns blank div on error 1`] = `<div />`;
